adagrams/game.py

```diff
@@ -1,11 +1,168 @@
+import random
+
+# Dictionary of the letters with their quantities
+LETTER_POOL = {
+    "A": 9,
+    "B": 2,
+    "C": 2,
+    "D": 4,
+    "E": 12,
+    "F": 2,
+    "G": 3,
+    "H": 2,
+    "I": 9,
+    "J": 1,
+    "K": 1,
+    "L": 4,
+    "M": 2,
+    "N": 6,
+    "O": 8,
+    "P": 2,
+    "Q": 1,
+    "R": 6,
+    "S": 4,
+    "T": 6,
+    "U": 4,
+    "V": 2,
+    "W": 2,
+    "X": 1,
+    "Y": 2,
+    "Z": 1,
+}
+
+# Dictionary assigning point values to the letters
+LETTERS_VALUE = {
+    "A": 1,
+    "B": 3,
+    "C": 3,
+    "D": 2,
+    "E": 1,
+    "F": 4,
+    "G": 2,
+    "H": 4,
+    "I": 1,
+    "J": 8,
+    "K": 5,
+    "L": 1,
+    "M": 3,
+    "N": 1,
+    "O": 1,
+    "P": 3,
+    "Q": 10,
+    "R": 1,
+    "S": 1,
+    "T": 1,
+    "U": 1,
+    "V": 4,
+    "W": 4,
+    "X": 8,
+    "Y": 4,
+    "Z": 10,
+}
+
+
 def draw_letters():
-    pass
+
+    # Draw 10 letters randomly, considering the quantity limits
+    letter_list = list(LETTER_POOL.keys())
+    tiles = {}  # Dictionary with letter and count to avoid nested loops
+    letter_bank = [] 
+
+    while len(letter_bank) < 10:
+        random_number = random.randint(0, len(letter_list) - 1)
+        random_letter = letter_list[random_number]
+
+        if random_letter in tiles:
+            if tiles[random_letter] >= LETTER_POOL[random_letter]:
+                continue
+            tiles[random_letter] += 1
+        else:
+            tiles[random_letter] = 1
+            
+        letter_bank.append(random_letter)
+
+    return letter_bank
+
+
+# Count occurrences of a letter in an iterable
+def count_items(item, iterable):
+    count = 0
+    for x in iterable:
+        if x == item:
+            count += 1
+    return count
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    capital_word = word.upper()
+
+    for letter in set(capital_word):
+        if letter not in letter_bank or count_items(letter, capital_word) > count_items(
+            letter, letter_bank
+        ):
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    # Calculate score
+    score = 0
+    for letter in word.upper():
+        score += LETTERS_VALUE[letter]
+
+    # Add bonus points for words of length 7-10
+    if len(word) >= 7 and len(word) <= 10:
```

Comment:
This looks good! Remember that we have a slightly more concise way to write a conditional to figure out if a value falls into some range. What might that look like?